ccf_transaction_processor/Readme.md

```diff
@@ -6,37 +6,142 @@ https://creativecommons.org/licenses/by/4.0/
+```
+
+A single node CCF instance hosting the PDO TP will be available for business use @ ip-address:6006. Here `ip-address` is the address of this VM that can be used to ping this machine from other VMs (this is our use case). Set ip-address to 127.0.0.1 for local testing. If you are behind a proxy, add ip-address to the list of no_proxy bash env variable.
+
+# Share CCF Keys
+
+CCF uses mutually authenticated TLS channels for transactions. Keys are located at
+$PDO_SRC/ccf_transaction_processor/CCF/build/workspace/pdo_tp_common. The network certificate is `networkcert.pem`.
+User public certificate is `user0_cert.pem` and private key is `user0_privk.pem`.
+
+In our usage, CCF users are PDO clients, and PDO client authentication is implemented within the transaction processor itself. Thus, we do not utilize the client authentication feature provided by CCF. However to satisfy the CCF's requirement that only authorized CCF users can submit transactions to CCF, share `user0_cert.pem` and `user0_privk.pem` with all the PDO clients. These two keys must be stored under $PDO_LEDGER_KEY_ROOT (as part of PDO deployment) under the names `userccf_cert.pem` and `userccf_privk.pem` respectively.
+
+# Test the Deployment with Ping Test
+
+PDO TP contains a simple `ping` rpc that returns success everytime it is queried. Test the PDO-TP deployment using this rpc. Invoke the following commands to issue 100 ping rpcs. The net througput is reported at the end of the test.
+
+```bash
+cd $PDO_SRC/ccf_transaction_processor/test
+./test.sh <ip-address>
+```
+
+The CCF port is assumed to be 6006.
+
+# Generate Signing Keys used to sign Payloads of Read Transactions
+
+Responses to read-transactions include a payload signature, where the signature is generated within PDO-TP.
+The required signing keys must be generated before PDO-TP can be opened up for business from PDO clients.
+Invoke the following commands to generate the signing keys.
+
+```bash
+cd $PDO_SRC/ccf_transaction_processor/generate_signing_keys
+./gen_keys.sh <ip-address>
+```
+
+If successful, the rpc returns after 1) creating one set of signing keys locally within the CCF enclave, and 2) scheduling
+them for global commit. The corresponding verifying key can be obtained using the `get_ledger_verifying_key`
+rpc. Verifying key is returned only after global commmit is successful. Invoke the following commands to
+check that signing keys have been globally committed before opening up CCF for business from PDO clients.
+
+```bash
+./get_verifying_keys.sh <ip-address>
+```
+
+The read-paylaod-signature feature may be used by PDO clients to establish offline verifiable proof of transaction commits as desired by the PDO smart contract application. Note that for trust purposes, it is recommended that any entity that uses the verifying_key gets it directly from the CCF service using the `get_ledger_verifying_key` rpc.
```

Comment:
What seems to miss in this doc is then also how to connect the CCF PDO to the PDO "front-end" (EHS, PS).  It is of course somewhat hinted in other documents (e.g., https://github.com/hyperledger-labs/private-data-objects/blob/master/docs/environment.md), but still would be nice to have a section how you could pull it together and, say, run a mock-contract and/or the standard tests?

Reply:
yes, agreed, I am adding a new section to the document on "Usage". Thank you!